src/DotNet/Resources/ResourceReader.cs

```diff
@@ -201,7 +201,8 @@ sealed class ResourceInfo {
 			if (typeIndex < 0 || typeIndex >= userTypes.Count)
 				throw new ResourceReaderException($"Invalid resource type index: {typeIndex}");
 			var type = userTypes[typeIndex];
-			switch (type.Name) {
+			var actualName = type.Name.Split(',')[0];
```

Comment:
It's better to use `count` if possible:

```suggestion
			var actualName = type.Name.Split(',', 2)[0];
```

Reply:
That overload doesn't exist :/ 
![image](https://user-images.githubusercontent.com/37494960/128498455-abafe7dc-3358-4520-b29f-131f56ac45d4.png)
The code would have to be:
`var actualName = type.Name.Split(new[] { ',' }, 2)[0];`
There is also the option of:
`string actualName = type.Name.Substring(0, type.Name.IndexOf(','));`

Reply:
Ok then (it seems like my method is available in later versions). The second option is not correct because `,` may be not presented. The following code is correct:

```cs
var commaIndex = type.Name.IndexOf(',');
string actualName = commaIndex == -1 ? type.Name : type.Name.Remove(commaIndex);
```